addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2368,6 +2368,70 @@ var SnippetsMenu = Widget.extend({
     _onSnippetSearchResetClick: function () {
         this._filterSnippets('');
     },
+    /**
+     * Adds the computation to the mutex queue and sets a loading effect over the editor after 1s.
+     * As soon as the mutex is unlocked, the loading effect will be removed.
+     * if the '_setLoadingEffect' has not already been executed, the 'mutexPromise' will be able
+     * to stop the execution by calling clearTimeout().
+     *
+     * @private
+     * @param {OdooEvent} ev
+     */
+    _execSnippetEditionRequest: async function (ev) {
+        this._mutex.exec(ev.data.exec);
+        const loadingTimer = setTimeout(() => {
+            this._setLoadingEffect();
+        }, 1000);
+        const mutexPromise = (timer) => {
+            return this._mutex.getUnlockedDef().then(() => {
+                clearTimeout(timer);
+                this._removeLoadingEffect();
+            });
+        };
+        await mutexPromise(loadingTimer);
```

Comment:
There is something strange with this code... reading it it seems that if I call this method 100x times, there will be 100 setTimeout running at the same time since you create one each time. I don't see the point of the `mutexPromise` function either: you define it, then you call it... so you could just do what it does directly.